end_to_end/plugin_test.go

```diff
@@ -322,6 +324,42 @@ var _ = Describe("End to End plugin tests", func() {
 					Skip("This test is only needed for the most recent backup versions")
 				}
 			})
+			It("Will not hang if gpbackup and gprestore runs with single-data-file and plugin fails at init for restore", func(ctx SpecContext) {
+				copyPluginToAllHosts(backupConn, examplePluginExec)
+
+				testhelper.AssertQueryRuns(backupConn, "CREATE TABLE t0(a int);")
+				testhelper.AssertQueryRuns(backupConn, "INSERT INTO t0 SELECT i FROM generate_series(1, 10)i;")
+				defer testhelper.AssertQueryRuns(backupConn, "DROP TABLE t0;")
+
+				output := gpbackup(gpbackupPath, backupHelperPath,
+					"--single-data-file",
+					"--plugin-config", examplePluginTestConfig)
+				timestamp := getBackupTimestamp(string(output))
+
+				backupCluster.GenerateAndExecuteCommand(
+					"Instruct plugin to fail",
+					cluster.ON_HOSTS,
+					func(contentID int) string {
+						return fmt.Sprintf("touch /tmp/GPBACKUP_PLUGIN_KILL_HELPER")
+					})
+
+				gprestoreCmd := exec.Command(gprestorePath,
+					"--timestamp", timestamp,
+					"--redirect-db", "restoredb",
+					"--plugin-config", examplePluginTestConfig)
+
+				_, err := gprestoreCmd.CombinedOutput()
+				Expect(err).To(HaveOccurred())
+
+				backupCluster.GenerateAndExecuteCommand(
+					"Unset plugin failure",
+					cluster.ON_HOSTS,
+					func(contentID int) string {
+						return fmt.Sprintf("rm /tmp/GPBACKUP_PLUGIN_KILL_HELPER")
+					})
```

Comment:
Will it make sense to put this into a defer block as in previous PR?

Reply:
Yes, updated